src/test/java/hw6/steps/ThenSteps.java

```diff
@@ -0,0 +1,89 @@
+    public void iAssertCheckboxLogsOnDifferentElementsPage(String checkboxOne, String checkBoxTwo) {
+        differentElementsPage = new DifferentElementsPage(WebDriverSingleton.INSTANCE.getDriver());
+        assertTrue(differentElementsPage.isCheckboxLogDisplayed(checkboxOne, "true"));
+        assertTrue(differentElementsPage.isCheckboxLogDisplayed(checkBoxTwo, "true"));
+    }
+
+    @Then("For radio button {string} there is a log row and value is corresponded to the status of radio button")
+    public void iAssertRadioLogsOnDifferentElementsPage(String radioName) {
+        assertTrue(differentElementsPage.isRadioLogDisplayed(radioName));
+    }
+
+    @Then("For dropdown value {string} there is a log row and value is corresponded to the selected value")
+    public void iAssertDropdownLogsOnDifferentElementsPage(String color) {
+        assertTrue(differentElementsPage.isDropdownLogDisplayed(color));
+    }
+
+    @Then("User Table page should be opened")
+    public void iAssertTablePageTitle() {
+        userTablePage = new UserTablePage(WebDriverSingleton.INSTANCE.getDriver());
+        assertEquals(userTablePage.getTitle(), "User Table");
+    }
+
+    @Then("{int} Number Type Dropdowns should be displayed on Users Table on User Table Page")
+    public void iAssertDropdownsOnUserTablePage(int expectedDropdownsAmount) {
+        assertEquals(userTablePage.getUsersSelects().size(), expectedDropdownsAmount);
+        assertTrue(userTablePage.isUsersDropdownsDisplayed());
+    }
+
+    @Then("{int} Usernames should be displayed on Users Table on User Table Page")
+    public void iAssertUsernamesOnUserTablePage(int expectedUsernamesAmount) {
+        assertEquals(userTablePage.getUsersNames().size(), expectedUsernamesAmount);
+        assertTrue(userTablePage.isUsersNamesDisplayed());
+    }
+
+    @Then("{int} Description texts under images should be displayed on Users Table on User Table Page")
+    public void iAssertDescriptionsUnderImagesOnUserTablePage(int expectedDescriptionsAmount) {
+        assertEquals(userTablePage.getUsersImagesDescriptions().size(), expectedDescriptionsAmount);
+        assertTrue(userTablePage.isUsersImagesDescriptionDisplayed());
+    }
+
+    @Then("{int} checkboxes should be displayed on Users Table on User Table Page")
+    public void iAssertCheckboxesOnUserTablePage(int expectedCheckboxesAmount) {
+        assertEquals(userTablePage.getUsersCheckboxes().size(), 6);
+        assertTrue(userTablePage.isUsersCheckboxesDisplayed());
+    }
+
+    @Then("User table should contain following values:")
+    public void iAssertContentOfTableOnUserTablePAge(DataTable dataTable) {
+        List<UserTableRow> actualUserTable = CalculationUtils
+                .convertToUserTableRowList(userTablePage.getUsersIds(), userTablePage.getUsersNames(), userTablePage.getUsersImagesDescriptions());
```

Comment:
Why `convertToUserTableRowList` is in the `CalculationUtils`? Also Cucumber has mechanism which allows set `List<UserTableRow>` as parameter to the step

Reply:
remade all that stuff with the help of POJO from params